### PR TITLE
Separate crypto and verify report modules from example/common/python and package

### PR DESCRIPTION
**Pull request by manju956**
this change is brought in to enable modularity in various avalon components.
Isolation of crypto and verify report and packaging them into a single entiry,
eliminates Avalon listener component dependecy on tc/sgx/common.

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: common/python/crypto_utils/setup.py [hunk 1]
```diff
@@ -133,9 +133,9 @@
 
 # -----------------------------------------------------------------
 setup(
-    name='tcf_examples_common',
+    name='tcf_crypto_utils',
     version=version,
-    description='Common library',
+    description='Crypto and Attestation verfication utils package',
```
Comment (danintel) [inline common/python/crypto_utils/setup.py]: s/verfication/verification/

Comment (manju956) [reply common/python/crypto_utils/setup.py]: Ok

**Comment by manju956** [comment]
> Approved, but fix these errors:
> 
> * s/Seperate/Separate/
> * s/verfication/verification/
> 
> Also, there is a build error.

I am looking into build issue.

**Comment by manju956** [comment]
> Approved, but fix these errors:
> 
> * s/Seperate/Separate/
> * s/verfication/verification/
> 
> Also, there is a build error.

**Comment by manju956** [comment]
> > Approved, but fix these errors:
> > 
> > * s/Seperate/Separate/
> > * s/verfication/verification/
> > 
> > Also, there is a build error.
> 
> I am looking into build issue.

build issue is fixed. Jenkins CI test is passed.

File: common/python/crypto_utils/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	@echo Build build_ext
+	python3 setup.py build_ext
+
+build :
+	mkdir $@
+	cd $@ 
+
+install:
+	@echo INSTALLING WHEEL FILE =================
+	pip3 install dist/tcf_crypto_utils*.whl
```
Comment (manojgop) [inline common/python/crypto_utils/Makefile]: tcf_crypto_utils -> avalon_crypto_utils

File: common/python/crypto_utils/setup.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    verify_report_modulefiles,
+    swig_opts=['-c++'] + ['-I%s' % i for i in verify_report_include_dirs],
+    extra_compile_args=compile_args,
+    include_dirs=verify_report_include_dirs,
+    library_dirs=library_dirs,
+    libraries=libraries)
+
+# -----------------------------------------------------------------
+setup(
+    name='tcf_crypto_utils',
```
Comment (manojgop) [inline common/python/crypto_utils/setup.py]: tcf_crypto_utils -> avalon_crypto_utils

File: examples/common/python/Makefile [hunk 4]
```diff
@@ -15,14 +15,9 @@
 PY_VERSION=${shell python3 --version | sed 's/Python \(3\.[0-9]\).*/\1/' | tr -d .}
 MOD_VERSION=${shell ../../../bin/get_version}
 
-# Format of wheel package is <pkg_name>-{python-tag}-{abi-tag}-{platform}
-WHEEL_FILE=dist/tcf_examples_common-${MOD_VERSION}-cp${PY_VERSION}-cp${PY_VERSION}m-linux_x86_64.whl
```
Comment (manojgop) [inline examples/common/python/Makefile]: Why is WHEEL_FILE variable removed ?

File: examples/common/python/Makefile [hunk 5]
```diff
@@ -36,10 +31,10 @@ build :
 
 install:
 	@echo INSTALLING WHEEL FILE =================
-	pip3 install $(WHEEL_FILE)
+	pip3 install dist/tcf_examples_common*.whl
```
Comment (manojgop) [inline examples/common/python/Makefile]: Better to use WHEEL_FILE variable.  PR https://github.com/hyperledger/avalon/pull/199 might remove modules outside examples/common/python. check with @Ram-srini

File: common/python/crypto_utils/Makefile (hunk 2 above)
Comment (manju956) [reply common/python/crypto_utils/Makefile]: Ok

File: common/python/crypto_utils/setup.py (hunk 3 above)
Comment (manju956) [reply common/python/crypto_utils/setup.py]: Ok

File: examples/common/python/Makefile (hunk 4 above)
Comment (manju956) [reply examples/common/python/Makefile]: wheel package naming convention is bit complex. Moreover, we really don't need suffix after the package name for install/uninstall using pip. One thing can be done is keeping the variable as below.
WHEEL_FILE=dist/tcf_example_common

File: examples/common/python/Makefile (hunk 5 above)
Comment (manju956) [reply examples/common/python/Makefile]: ok

File: examples/common/python/Makefile (hunk 4 above)
Comment (manojgop) [reply examples/common/python/Makefile]: https://www.python.org/dev/peps/pep-0427/#file-name-convention. Let's follow existing whl file namng convention.  Format of wheel package is <pkg_name>-{python-tag}-{abi-tag}-{platform}

Comment (manju956) [reply examples/common/python/Makefile]: ok

File: common/python/crypto_utils/setup.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    version=version,
+    description='Crypto and Attestation verification utils package',
+    author='Intel',
+    url='http://www.intel.com',
```
[the remark this answers is not in the document]
Comment (manju956) [reply common/python/crypto_utils/setup.py]: Ok

**Comment by manju956** [comment]
SGX dependency on untrusted part of crypto and verify report are eliminated as part of https://github.com/hyperledger/avalon/pull/204 and https://github.com/hyperledger/avalon/pull/203.
There is no need for these changes, hence closing the PR.

**Comment by manju956** [comment]
SGX dependency on untrusted part of crypto and verify report are eliminated as part of #204 and #203.
There is no need for these changes, hence closing the PR.

**Pull request closed without merging**